### PR TITLE
Add more details on PR process to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jkwak-work**
This commit adds more detailed explanations on the PR process.
It also fixed a few typos and grammar mistakes identified by LLM.

Closes https://github.com/shader-slang/slang/issues/5730

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -226,31 +269,50 @@ $ git submodule update --recursive
...
-1. Don't use the STL containers, iostreams, or the built-in C++ RTTI system.
-1. Don't use the C++ variants of C headers (e.g., use `<stdio.h>` instead of `<cstdio>`).
-1. Don't use exceptions for non-fatal errors (and even then support a build flag to opt out of exceptions).
-1. Types should use UpperCamelCase, values should use lowerCamelCase, and macros should use `SCREAMING_SNAKE_CASE` with a prefix `SLANG_`.
-1. Global variables should have a `g` prefix, non-const static class members can have an `s` prefix, constant data (in the sense of static const) should have a `k` prefix, and an `m_` prefix on member variables and a `_` prefix on member functions are allowed.
-1. Prefixes based on types (e.g., p for pointers) should never be used.
-1. In function parameter lists, an `in`, `out`, or `io` prefix can be added to a parameter name to indicate whether a pointer/reference/buffer is intended to be used for input, output, or both input and output.
-1. Trailing commas should always be used for array initializer lists.
-1. Try to write comments that explain the "why" of your code more than the "what".
+2. Don't use the STL containers, iostreams, or the built-in C++ RTTI system.
```
Comment (csyonghe) [inline CONTRIBUTING.md:307]: Using "1." is proper markdown syntax so we don't have to manually keep track of the numbering.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -97,127 +100,167 @@ $ sudo apt-get install cmake ninja-build
...
+2. Source code "Format" needs to be changed.
+
+### Addressing Code Reviews
+After your pull request is created, you will receive code reviews from the community within 24 hours.
+
+The PR requires approval from people who have permissions. They will review the changes before approving the pull. During this step, you will get feedback from other people, and they may request you to make some changes.
+
+Follow-up changes that address review comments should be pushed to your pull request branch as additional commits. Any additional commits made to the same branch in your forked repository will show up on the PR as incremental changes.
 
+When your branch is out of sync with top-of-tree, submit a merge commit to keep them in sync. Do not rebase and force push after the PR is created to keep the change history during the review process.
```
Comment (csyonghe) [inline CONTRIBUTING.md:261]: Add:
"The Slang repository is configured to use "squashing" strategy when merging pull requests. Commits accumulated in your PR will be automatically squashed into a single commit to top of tree when it lands."

**Review by csyonghe — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (jkwak-work) [reply CONTRIBUTING.md:261]: This is already on the document at line 270.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jkwak-work) [reply CONTRIBUTING.md:307]: Right. I will bring it back.

**Pull request merged**